src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -36,35 +42,39 @@ abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnal
       .endpoint(getEndpoint)
       .buildClient()
 
+  import spark.implicits._
+  lazy val df: DataFrame = Seq(
+    (Seq(""),Seq("")),
+  ).toDF("lang", "text")
+
   protected def transformTextRows(toRow: TAResponseV4[T] => Row)
                                  (rows: Iterator[Row]): Iterator[Row] = {
+
     rows.map { row =>
       val results = invokeTextAnalytics(getValue(row, text), getValue(row,lang))
       Row.fromSeq(row.toSeq ++ Seq(toRow(results))) // Adding a new column
     }}
 
   override def transform(dataset: Dataset[_]): DataFrame = {
+    val batchedDF = new FixedMiniBatchTransformer().setBatchSize(10).transform(dataset.coalesce(1))
+    val finaldataset = spark.createDataFrame(batchedDF.rdd, df.schema)

```

Comment:
I would suggest instead of creating a temp df here, creating the schema directly: 

https://sparkbyexamples.com/spark/spark-schema-explained-with-examples/

  val inputSchema = StructType(Array(
    StructField("lang",ArrayType(StringType,true),true),
    StructField("text",ArrayType(StringType,true),true)
  ))